### ISSUE TITLE
✔️ 🚀 Feature Request: Allow to add Multiple rows in one section

### ISSUE THREAD

**Issue by black-eye** [elementor/elementor#2218]
It would be nice to have a possibility to add multiple rows in one section. It would be useful for situations like these:
![elementor-rows](https://user-images.githubusercontent.com/5634870/29495011-79e835a2-85b6-11e7-80ee-c1c7a57f4f3a.png)

This is actually possible to do in tablet and mobile view, where I can set whatever column length I want, and if the sum of all columns lengths is more than 100%, the spare columns start on a new row. This approach is impossible on desktop though, because the sum of all columns is allways 100%.

Maybe it would be sufficient to have an on/off toggle on this allways-100%-width rule.

**Comment by vongoh** [comment]
+1

**Comment by mmjaeger** [comment]
+1

**Comment by enviedbymost** [comment]
+1

**Comment by fthiesen** [comment]
+1

**Comment by maxwellhibbert** [comment]
Someone had the idea of adding a column wrap setting that would allow columns to exceed 100% on desktop. Much needed.

**Comment by ramonztro** [comment]
+1

**Comment by GostkaDesign** [comment]
for sure maybe you can inpire you by keith clark code for make it simple
https://codepen.io/keithclark/pen/xVvpam

**Comment by black-eye** [comment]
Any progress here? At least an option to enable exceeding 100% width on the desktop?
Should be quite an easy task, shouldn't it? It works on the mobile and tablet just fine, I just need it on the desktop as well. 

At least any advice how to "hack" it, e.g. via filter, action or custom code?

**Comment by drank** [comment]
+1

**Comment by jonlewis14** [comment]
+1

**Comment by sourabhgarg01** [comment]
+1

**Comment by batnyam** [comment]
+1

**Comment by CasperEngl** [comment]
+1

**Comment by EldoRay** [comment]
+1

**Comment by laura-ionescu** [comment]
+1

**Comment by ArekZw** [comment]
+1

**Comment by AngeloMaciel** [comment]
+1

Would be very useful for nesting title and other section related elements.

**Comment by tulensytyttaja** [comment]
+1

But there actually is a way around this issue, if you just place all the columns in one row and then give a CSS-id for the parent element where you have the row and then just add a little custom CSS to one of the elements or where ever you add your CSS.

```
#idyougavetoparentelement .elementor-row {
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
}

#idyougavetoparentelement .elementor-top-column {
    width: 33% !important;
}

@media (max-width: 1025px) {
#idyougavetoparentelement .elementor-top-column {
    width: 50%;
  }
}

@media (max-width: 700px) {
#idyougavetoparentelement .elementor-top-column {
    width: 100%;
  }
}

```

**Comment by black-eye** [comment]
Anything new on this one? I'm starting to be quite desperate on this one.
Could you (anybody) please at least direct me to a place in the code where the condition for the max 100% width rule is? To turn it off manually. I'm desperate enough to edit the source code and change it every time new update comes...
Thanks

**Comment by NickJCUK** [comment]
I really need this "multiple rows in one section" - in fact, multple "anthing" in one section would be good!

**Comment by tomas-eklund** [comment]
+1

I did something similar to @tulensytyttaja. On the Advanced Tab of the section containing the columns I added the following Custom CSS:
```
@media (min-width: 1024px) {
    selector .elementor-row {
        display: block;
    }
    selector .elementor-column {
        width: 33%;
        display: block;
        float: left;
    }
}
```
I did not write custom CSS for the tablet and mobile layouts as I was able to set these widths using the Elementor GUI. But being able to do it without custom CSS would certainly be a lot better. You still need to be a coder to use Elementor effectively.

**Comment by notrealdev** [comment]
Need this feature

**Comment by johnlewissims** [comment]
+1

Another feature Elementor really needs.

**Comment by m2sahin** [comment]
+1

**Comment by reardestani** [comment]
Faced this limitation today, guys please add this feature as soon as possible.

**Comment by kubik101** [comment]
+1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1 +1  .. please :)

**Comment by kubik101** [comment]
This worked for me.
`.my_parent_100 .elementor-row {
	display: flex;
	flex-wrap: wrap;
	.my_100 {
		width: 100%;
	}
}`

**Comment by laratuk** [comment]
Add this Custom CSS for Section and use PAFE Column Width in Piotnet Addons to do this. 
Video Tutorial : https://drive.google.com/file/d/1pdUFGXgZdZ_eD8IrMXGfsZip9b3g_Nk_/view?usp=sharing

selector .elementor-row {
    flex-wrap: wrap;
}

**Comment by philmatt** [comment]
+1

**Comment by MPHood** [comment]
+1+1+1+1

**Comment by land0r** [comment]
+1

**Comment by tictok** [comment]
+1 Definitely.
Can currently achieve this using custom CSS but visually doesn't look great in the editor, so something like a setting for a custom 'structure' where flex-wrap can be toggled, and desired number of items per row selected (or item percentage), for each breakpoint would be great! 

( I know that currently within a 'row / structure' items can have their width percentage set individually, but combined elements can only add up to 100% and get squished into the same row).

**Comment by egortsaryk9** [comment]
Why it's still not implemented ?

**Comment by kowsar89** [comment]
+1

**Comment by jotcreative** [comment]
+1

**Comment by geokongr** [comment]
Very useful feature. Interestingly enough, tablet and mobile views support columns that extend in multiple rows, why not Desktop view?

The workaround i suggest is to use 2 rows with different visibility: The first row will contain two inner rows (inner sections) with 3 columns each and will have Desktop visibility (1st screenshot). The second row will contain 3 inner rows with 2 columns each with will have Tablet and Mobile visibility (2nd screenshot).

This may probably add performance overhead and definitely adds design complexity so i strongly suggest to implement the feature request.
![Screenshot 1  - 3x2 grid](https://user-images.githubusercontent.com/29377122/73406337-e0458400-42fe-11ea-90e6-f99657320a46.jpg)
![Screenshot 2  2x3 grid](https://user-images.githubusercontent.com/29377122/73406346-e76c9200-42fe-11ea-8b05-a4557c6ccf1f.jpg)

**Comment by lukeaaron** [comment]
Further to @tomas-eklund and @tulensytyttaja's solution, in the case of columns with different heights – that can cause columns in the rows below to float in seemingly unexpected positions – I would add some further CSS to clear after each row.

@media (min-width: 1024px) {
    selector .elementor-row {
        display: block;
    }
    selector .elementor-column {
        width: 33%;
        display: block;
        float: left;
    }
    **selector .elementor-column:nth-child(3n+4) {
        clear: both;
    }**
}

This uses nth-child CSS functionality to clear every third column (as there are three columns per row), starting from the second row's first column (the fourth column).

PS. You can adjust the CSS function accordingly, if you were to want to show more than three columns per row on desktop – noting you would want to change the percentage in the "selector .elementor-column" style above it also.

**Comment by BrockwayProd** [comment]
+1 support for flexible column/rows on desktop view

**Comment by ErwinLiemburg** [comment]
+1 from me too

**Comment by yossicahen** [comment]
+1

**Comment by arun369** [comment]
+1

Any update on this by ant chance? 
This will make Elementor more awesome.

**Comment by MichaelDarko** [comment]
+ 1 This would be very useful.

**Comment by MichaelDarko** [comment]
I'm also missing this feature pretty badly. We have flexbox so Elementor should allow us to wrap columns on desktop, not only tablet or mobile.

**Comment by arekuusa** [comment]
+1

**Comment by roboxtim** [comment]
+1 pls

**Comment by Stml-St** [comment]
+1

**Comment by workshop22** [comment]
+1

**Comment by black-eye** [comment]
Jeez, I don't get this! It's been more than 3 years since this issue was opened!

All it has to be done is to add some kind of little switch to make the desktop settings (optionally) behave the same way as the tablet and mobile. What is so hard about it, that it takes more than 3 years?

@arielk, @shilo-ey, @udidol, @joshmarom, @bainternet, anyone?

**Comment by deniskern** [comment]
+1

**Comment by rawk7000** [comment]
+1

**Comment by ijjimem** [comment]
One of the easiest to implement yet no advancement ?

**Comment by tomas-eklund** [comment]
Needed.

**Comment by alicansoysal** [comment]
+1

**Comment by alicansoysal** [comment]
+1

this is the only deficiency compared to wpbakery.

**Comment by anjanphukan** [comment]
+1
I need this too.

**Comment by addMitt** [comment]
As a web developer who just started using this software, very disappointed this isn't possible.

Really, a lot of this stuff could likely be achieved with nested Inner Sections, especially if you allow us to make them inline. Even better if we had full control over the flex settings from a Section.

**Comment by Stml-St** [comment]
+1

**Comment by nicpie** [comment]
+1.....!!!!

**Comment by luisceladita** [comment]
Come guys, This is so easy to implement

**Comment by nufrankz** [comment]
Multiple rows inside a section (using Elementor's widgets). We need this if isn't created yet, thanks.

**Comment by Chris29888** [comment]
Holy s..., how is this still not a thing?! I'm so tired of Elementor blocking my progress at every possible step. Accidentally bought a theme with this piece of junk pre-installed and now I'm stuck working on a fairly large project and I'm losing my mind cause I'm constantly running into these annoying little issues that take hours to "figure out". "Figuring out" usually means realizing it's not possible and finding a workaround.
Might as well use Squarespace, Wix or one of the other pleb page builder providers.

**Comment by 79ho3ein** [comment]
+1

**Comment by Camilo517** [comment]
+1

**Comment by ImNightwing** [comment]
+1 ............................

**Comment by birquah** [comment]
How has this not been resolved? Uikit and Bootstrap had this for years before this thread was created.  Someone on the Elementor team, please help!

**Comment by amstec** [comment]
I have a work around. Create a section with a single column then add rows to that column.

<img width="1074" alt="Screen Shot 2021-07-30 at 1 47 27 PM" src="https://user-images.githubusercontent.com/6069527/127706292-a2596616-8494-41c5-ae46-6ceec21d1129.png">

**Comment by birquah** [comment]
@amstec this will not work if you have an odd amount of items.

**Comment by 79ho3ein** [comment]
> I have a work around. Create a section with a single column then add rows to that column. No CSS or coding changes required
> 
> <img alt="Screen Shot 2021-07-30 at 1 47 27 PM" width="1074" src="https://user-images.githubusercontent.com/6069527/127706292-a2596616-8494-41c5-ae46-6ceec21d1129.png">

This method is not optimal and generates a lot of code.

**Comment by amstec** [comment]
Part of the original request was "It would be nice to have a possibility to add multiple rows in one section." This is a way to do this in the current release. Yes I agree it is not ideal, but it is a possible solution till a real one is implemented if someone is looking for something similar.

**Comment by birquah** [comment]
It is not a method that works, though. Using your example if you have 2 rows of three columns, when you go to tablet breakpoint you will en up with two sets of two columns in one row and one in the next. What we need is the ability to say we have 6 items that need to be in a responsive box that adjusts width on the item and not the row. That way the 6 items could be in two rows of three then pop down to three rows of two on tablet size.

**Comment by Pumpk1in** [comment]
So, 2021 and still forced to use CSS hacks ?
Jesus.

**Comment by vgstef** [comment]
Yeah, I also regularly need this feature. Here what I do :

I create a section with only only column.

Then I use inner section with this custom CSS :
```
selector {
    display: inline;
    max-width: 300px;
}
```
This way, I have "wrapped" columns.

Elementor could simply and Positioning options in the Advanced tab for inner section, just like we find for other widgets.

**Comment by 79ho3ein** [comment]
> Yeah, I also regularly need this feature. Here what I do :
> 
> I create a section with only only column.
> 
> Then I use inner section with this custom CSS :
> 
> ```
> selector {
>     display: inline;
>     max-width: 300px;
> }
> ```
> 
> This way, I have "wrapped" columns.
> 
> Elementor could simply and Positioning options in the Advanced tab for inner section, just like we find for other widgets.

This feature will be added in elementor 3.6. Version 3.6 brings tremendous changes.

**Comment by fabianenzensperger** [comment]
+1000000 (but i know it's coming in 3.6 with the new flexbox containers)

**Comment by ajaypatelaj** [comment]
+9999 Indeed needed one

**Comment by IT-Mesteren** [comment]
This would definitely improve elementor beyond anything!

**Comment by fabianenzensperger** [comment]
> This would definitely improve elementor beyond anything!

Why not using the new containers? They can do this.

**Comment by nicholaszein** [comment]
Hello friends!

We have great news! 

### 📢 We're happy to announce that you can now nest as many containers as you want using the now <ins>stable</ins> version of the Flexbox Container experiment released with Elementor **v3.8**! 🥳 

This means you can have as many rows as you want, with as many columns as you want. 🙌 

I encourage everyone to check it out and update your plugin to the new version!

Cheers 🥂
